File: Euclidean_algorithm/PHP/jakobhans/euclidean.php
<?php
    function gcd_division($a, $b)
    {
        while ($b != 0) {
            $tmp = $b;
            $b = $a % $b;
            $a = $tmp;
        }
        return $a;
    }

    function gcd_substraction($a, $b)
    {
        while ($a != $b) {
            if ($a > $b) {
                $a = $a - $b;
            } else {
                $b = $b - $a;
            }
        }
        return $a;
    }

    function gcd_recursive($a, $b)
    {
        if ($b == 0) {
            return $a;
        } else {
            return array($b, $a % $b);
        }
    }

File: Euclidean_algorithm/Ruby/mitogh/euclidean.rb
def gcd_division(a, b)
  while b != 0 do
    a, b = b, a % b
  end
  a
end

def gcd_substraction(a, b)
  while(a != b) do 
    a = a - b if a > b
    b = b - a if a < b
  end
  a
end

def gcd_recursive(a, b) 
  (b == 0) ? a : gcd_recursive(b, a % b)
end
